src/plugins/presentation_util/public/components/floating_actions/floating_actions.tsx

```diff
@@ -5,29 +5,81 @@
  * in compliance with, at your election, the Elastic License 2.0 or the Server
  * Side Public License, v 1.
  */
-import React, { FC, ReactElement } from 'react';
-
+import React, { FC, ReactElement, useEffect, useState } from 'react';
+import { TypedUseSelectorHook } from 'react-redux';
 import classNames from 'classnames';
+
+import { IEmbeddable, panelHoverTrigger, PANEL_HOVER_TRIGGER } from '@kbn/embeddable-plugin/public';
+import { Action } from '@kbn/ui-actions-plugin/public';
+
+import { pluginServices } from '../../services';
 import './floating_actions.scss';
+import { ReduxEmbeddableState } from '../../redux_embeddables';
 
 export interface FloatingActionsProps {
-  className?: string;
-  actions?: JSX.Element;
   children: ReactElement;
+  select: TypedUseSelectorHook<ReduxEmbeddableState>;
+
+  className?: string;
   isEnabled?: boolean;
+  embeddable?: IEmbeddable;
 }
 
 export const FloatingActions: FC<FloatingActionsProps> = ({
-  className = '',
-  actions,
-  isEnabled,
+  select,
   children,
+  isEnabled,
+  embeddable,
+  className = '',
 }) => {
+  const {
+    uiActions: { getTriggerCompatibleActions },
+  } = pluginServices.getServices();
+
+  const viewMode = select((state) => state.explicitInput.viewMode);
+  const disabledActions = select((state) => state.explicitInput.disabledActions);
```

Comment:
I tried to pull as much logic as possible into the `FloatingActions` component to keep the consumers as clean as possible - this meant adding the action fetching logic to this component. However, this caused an issue where I needed things to be re-fetched when certain control group state changed - for example, view mode or the disabled actions. 

To get around this, I am taking in the Redux state selector hook so that the `FloatingActions` component can re-fetch the compatible actions as these parts of state change... I tried a few different things, including taking each piece of state as a prop instead, but this seemed the cleanest to me? It ties the `FloatingActions` component to the `ReduxEmbeddableState` though, so I'm open to other ideas :eyes:

Reply:
I think ideally we wouldn't want the floating actions tied to any sort of redux-specific state. Most embeddable containers won't use that I think. Is there any complexity in just passing in `viewMode` and `disabledActions` as props?